pages/index.vue

```diff
@@ -106,66 +152,134 @@
+          this.loading = false
+        })
+        .catch((err) =>
+          $toast.error(
+            "Can't fetch data right now. Please reload to try again."
+          )
+        )
+    },
+
+    updateTask(_index, _id) {
+      var newTodo = this.todos[_index].updatedValue
+
+      if (newTodo === '') {
+        this.$toast.error('Todo value cannot be blank. ')
+        return
+      }
+
+      if (newTodo === this.todos[_index].title) {
+        this.$toast.info('No change in the value of the todo.')
+        this.editTask(_index)
+        return
+      }
+
+      const headersForAuthorization = {
+        Authorization: 'Token ' + this.$store.getters.token,
+      }
+      const dataForAPIRequest = {
+        title: newTodo,
+      }
+
+      this.$toast.info('Updating your todo....')
+      this.$axios({
+        headers: headersForAuthorization,
+        url: 'todo/' + _id + '/',
+        method: 'patch',
+        data: dataForAPIRequest,
+      })
+        .then(() => {
+          this.$toast.success('Todo updated successfully!')
+          this.todos[_index].title = newTodo
+          this.editTask(_index)
+        })
+        .catch(() =>
+          this.$toast.error('Updation failed. Please try again later.')
+        )
     },
-    /**
-     * Function to update a single todo
-     * @argument {number} _index - index of element to update in todos array
-     * @argument {number} _id - id of todo obtained from API
-     * @todo Complete this function.
-     * @todo 1. Send the request to update the task to the backend server.
-     * @todo 2. Update the task in the dom.
-     */
-    updateTask(_index, _id) {},
-    /**
-     * toggle visibility of input and buttons for a single todo
-     * @argument {number} index - index of element to toggle
-     * @todo add in bindings in dom so that 'hideme' class is dynamic or use conditional rendering
-     * @hint read about class bindings in vue
-     */
-    editTask(index) {
-      this.todos[index].editing = !this.todos[index].editing
+
+    editTask(_index) {
+      this.todos[_index].editMode = !this.todos[_index].editMode
     },
-    /**
-     * Function to delete a single todo
-     * @argument {number} _index - index of element to update in todos array
-     * @argument {number} _id - id of todo obtained from API
-     * @todo Complete this function.
-     * @todo 1. Send the request to delete the task to the backend server.
-     * @todo 2. Remove the task from the dom.
-     */
-    deleteTask(_index, _id) {},
+
+    deleteTask(_index, _id) {
+      const headersForAuthorization = {
+        Authorization: 'Token ' + this.$store.getters.token,
+      }
+
+      this.$toast.info('Deleting your todo....')
+      this.$axios({
+        headers: headersForAuthorization,
+        url: 'todo/' + _id + '/',
+        method: 'delete',
+      })
+        .then(() => {
+          this.$toast.success('Todo deleted successfully!')
+          this.todos.splice(_index, 1)
```

Comment:
good use of splice

Reply:
Filtered Todos not updated

Reply:
I just realized this. Any idea why is this happening?
I am updating `this.todos` as using the `splice` method, and the `filteredTodos` is itself a computed property defined on `this.todos`, so shouldn't it be updating automatically?